### PR TITLE
Feature/view user profile

### PR DESCRIPTION
**Pull request by damola-sd**
# Description

Implemented view userProfile and wrote tests

## Checklist

Remove any items which are not applicable.

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works AND the tests pass

### CODE REVIEW

File: api/controllers/user.js [hunk 1]
```diff
@@ -9,14 +9,18 @@ const jwt = require('../helpers/jwt');
...
       });
-      if (user) {
-        return response.success(res, 201, user);
-      }
-      return response.error(res, 404, 'User with that username does not exist');
+
+      return response.success(res, 200, user);
     } catch (error) {
       return response.error(res, 500, error.message);
     }
```
Comment (iamcristos) [inline api/controllers/user.js]: what if there are no users existing????

File: api/controllers/user.js [hunk 2]
```diff
@@ -9,14 +9,18 @@ const jwt = require('../helpers/jwt');
...
+            as: 'location'
+          }
+        ]
       });
-      if (user) {
-        return response.success(res, 201, user);
-      }
-      return response.error(res, 404, 'User with that username does not exist');
+
+      return response.success(res, 200, user);
```
Comment (iamcristos) [inline api/controllers/user.js]: what if there are no users existing

File: __test__/users.test.js [hunk 3]
```diff
@@ -145,33 +145,43 @@ describe('GET /users', () => {
...
+      });
+  });
+
+  it('should return 401 if no token is provided', () => {
+    return request(server)
+      .get('/api/user/john/profile')
+      .then(res => {
+        expect(res.status).toBe(401);
+        expect(res.body.message).toBe('Token is required');
+      });
```
Comment (iamcristos) [inline __test__/users.test.js]: can you also write the test that validates that the user providing the token is the user accessing it????
check-in the helper folder the jwt file and auth user function

Comment (damola-sd) [reply __test__/users.test.js]: Wrote the tests

File: api/controllers/user.js (hunk 2 above)
Comment (damola-sd) [reply api/controllers/user.js]: Fised this also

**Pull request merged**